File: cs2041/lab05/echon.pl
#!/usr/bin/perl -w

if ($#ARGV+1 < 2) {
    print STDERR "Usage: $0 Not enough arguments\n";
} elsif ($#ARGV+1 > 2) {
    print STDERR "Usage: $0 Too many arguments\n";
} else {
    if($ARGV[0] =~ /^\d+$/) {
        foreach (1..$ARGV[0]) {
            print "$ARGV[1]\n";
        }
    } else {
        print STDERR "Usage: $0\n First argument should be a positive integer";
    }
}

File: cs2041/lab06/echon.py
#!/usr/bin/python

import sys, re

if len(sys.argv) == 1:
    sys.stderr.write("Usage: " + sys.argv[0] + " Not enough arguments")
    exit()
elif len(sys.argv) > 3:
    sys.stderr.write("Usage: " + sys.argv[0] + " Too many arguments")
    exit()
elif not re.match(r'^[0-9]+$', sys.argv[1]):
    sys.stderr.write("UsageL " + sys.argv[0] + " First argument must be a number")
    exit()

for i in range(0, int(sys.argv[1])):
    print sys.argv[2]
